### PR TITLE
Edit Developer Guide

### PR DESCRIPTION
**Pull request by Ryan-Tan00**
* Consolidated all `get` commands and changed the sequence diagrams to show `get` commands that require parameters and `get` commands that only require a prefix
* Updated user story for getting past appointments
* Updated Logic component to reflect the logic flow of `get` commands

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -170,8 +170,13 @@ How the parsing works:
...
   a `Command` object.
-* All `XYZCommandParser` classes (e.g., `AddCommandParser`, `DeleteCommandParser`, ...) inherit from the `Parser` 
-  interface so that they can be treated similarly where possible e.g, during testing.
+* When parsing a `get` command, the `AddressBookParser` class creates a `GetCommandParser` to parse the prefix of the 
+  `get` command (e.g., `/hw`). If the `get` command only requires a prefix (e.g., `get /inp` & `get /outp`), the 
+  respective `GetXYZCommand` object is created. If the `get` command requires parameters (e.g., `get /hw North`), the
+  respective `GetXYZCommandParser` is created to parse the parameters and create the appropriate `GetXYZCommand` to be
+  returned.
+* All `XYZCommandParser` and `GetXYZCommandParser` classes (e.g., `AddCommandParser`, `DeleteCommandParser`, ...) 
+  inherit from the `Parser` interface so that they can be treated similarly where possible e.g, during testing.
```
Comment (nehcuy) [inline docs/DeveloperGuide.md:179]: I think a better way to phrase it when our `get` commands require parameters is:
```
If the `get` command requires parameters (e.g., `get /hw North`), the prefix is parsed accordingly
within the `GetCommandParser` before the respective `GetXYZCommandParser` is created to
parse the parameters and create the appropriate `GetXYZCommand` to be returned.
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -348,20 +362,29 @@ Getting the list of inpatients and outpatients involves the following steps:
...
    to be returned in a `CommandResult`
 5. The list of `PastAppointment` will then be displayed in the `ResultDisplay`
 
+#### Getting patients with an appointment on a specified date (`get /appton`)
+
+Getting patients with an appointment on a specified date involves the following steps:
+1. prefix `/appton` is matched using an instance of `GetCommandParser`
+2. a new `GetAppointmentByDateCommandParser` instance is created and parses the user input (specifically the date inputted)
+3. a `GetAppointmentByDateCommand` instance containing the date of the appointment is created and returned
+4. the `GetAppointmentByDateCommand` command is executed, accessing the list of `PastAppointment` of the specified patient
```
Comment (nehcuy) [inline docs/DeveloperGuide.md:381]: Perhaps the word "command" following `GetAppointmentByDateCommand` can be removed for consistency with point 4 of `get /appt`.

**Review by nehcuy — CHANGES_REQUESTED** [review]
Just a quick change to the DG and it's GTM!

**Review by kxrt — APPROVED** [review]
Thanks for applying the requested changes!

**Review by nehcuy — APPROVED** [review]
Thanks for making the relevant changes! LGTM now.

**Pull request merged**